plugins/migration/bin/rename.js

```diff
@@ -0,0 +1,286 @@
+#!/usr/bin/env node
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview A script that automatically upgrades pieces of the Blockly API
+ *     that have been renamed by modifying developers' local files.
+ */
+'use strict';
+
+import {createAndAddSubCommand} from './command.js';
+import fetch from 'node-fetch';
+import glob from 'glob';
+import {readFile, writeFile} from 'fs';
+import semver from 'semver';
+import JSON5 from 'json5';
+
+const DATABASE_URL = `https://raw.githubusercontent.com/google/blockly/develop/scripts/migration/renamings.json5`;
+
+/**
+ * The version associated with renamings that have not been released yet.
+ * @const {string}
+ */
+const DEV_VERSION = 'develop';
+
+createAndAddSubCommand(
+    'rename', '>=5', 'Perform renamings for breaking changes')
+    .option(
+        '-i [suffix]',
+        'do renamings in-place, optionally create backup files with the ' +
+        'given suffix')
+    .action(async function() {
+      const fromVersion = this.processedArgs[0];
+      const toVersion = this.processedArgs[1];
+      const fileGlobs = this.processedArgs[2];
+      const fileNames = fileGlobs.flatMap((fileGlob) =>
+        glob.sync(fileGlob, {nodir: true, nonull: false}));
+
+      if (!fileNames.length) {
+        console.log(`No matching files found for ${fileGlobs}. ` +
+            `Aborting rename.`);
+        return;
+      }
+
+      const renamer = new Renamer(await getDatabase(), fromVersion, toVersion);
+      fileNames.forEach((name) => {
+        readFile(name, 'utf8', (err, contents) => {
+          if (err) throw err;
+          const newContents = renamer.rename(contents);
+          const i = this.opts().i;
+          if (i) {
+            if (typeof i == 'string') writeFile(name + i, contents, throwError);
+            writeFile(name, newContents, throwError);
+            console.log(`Migrated renamings in ${name}`);
+          } else {
+            process.stdout.write(newContents);
+          }
+        });
+      });
+    });
+
+/**
+ * Throws the error if the error is received, otherwise noop.
+ * @param {!Error|null} err A possible error to throw.
+ */
+function throwError(err) {
+  if (err) throw err;
+}
```

Comment:
Do the errors thrown by this function contain an informative stack trace?  Do they cause the program to terminate?